### PR TITLE
Fix "".format()

### PR DESCRIPTION
**Pull request by awildtechno**
Updated the implementation and test for new-style string formatting so that most tests pass. There is one remaining issue: Floating points are being rounded by the test suite, meaning {e} formats on floating-point numbers display incorrectly in the test suite. Someone else seems to have found a fix for that, so I should be able to take care of it as soon as I get more details.

All other pending issues appear to be fixed.

## PR Checklist:

- [ x ] All new features have been tested
- [ x ] All new features have been documented (no new features)
- [ x ] I have read the **CONTRIBUTING.md** file
- [ x ] I will abide by the code of conduct

### CODE REVIEW

File: tests/datatypes/test_str.py [hunk 1]
```diff
@@ -1438,13 +1438,18 @@ def test_conversion_types(self):
...
+                    except ValueError as err:
+                        print(err)
+                    except OverflowError as err:
+                        print(err)
                     """.format(alter=alter, typ=typ, arg=arg)
                 ) for alter, typ, arg in combinations
             ]
         )
-
+
```
Comment (brutusthebee) [inline tests/datatypes/test_str.py]: At column 1: (W293) blank line contains whitespace

File: package-lock.json [hunk 2]
```diff
@@ -17,7 +17,6 @@
       "version": "7.2.2",
```
Comment (martica) [inline package-lock.json]: You don't want to include this in the PR.

File: tests/datatypes/test_str.py (hunk 1 above)
Comment (martica) [reply tests/datatypes/test_str.py]: There are a few other blank lines with whitespace. Not sure why brutus didn't complain, but they generally shouldn't be there.

File: batavia/types/Str.js [hunk 3]
```diff
@@ -731,9 +731,28 @@ Str.prototype.capitalize = function() {
...
-    const positionalArguments = new types.Tuple(types.js2py(args))
-    const keywordArguments = types.js2py(kwargs)
+    var positionalArguments = undefined
+
+    // Convert args to python
+    args = types.js2py(args)
+    if (args.__iter__ || args.__getitem__ && args.__len__) {
+        // We were passed an iterable.
+        positionalArguments = new types.Tuple(args)
+    } else {
```
Comment (martica) [inline batavia/types/Str.js]: Does this else path happen? args should always be a JS list.

**Comment by awildtechno** [comment]
Closing for #806

**Pull request closed without merging**